src/integrationTest/java/uk/gov/hmcts/reform/ccd/documentam/controller/CaseDocumentAmControllerIT.java

```diff
@@ -0,0 +1,375 @@
+    private static final String USER_ID = "d5566a63-f87c-4658-a4d6-213d949f8415";
+
+    private static final String SERVICE_AUTHORISATION_KEY = "ServiceAuthorization";
+    private static final String BEARER = "Bearer ";
+    private static final String TOKEN = "eyJhbGciOiJIUzUxMiJ9.eyJzdWIiOiJjY2RfZ3ciLCJleHAiOjE1ODM0NDUyOTd9"
+        + ".WWRzROlKxLQCJw5h0h0dHb9hHfbBhF2Idwv1z4L4FnqSw3VZ38ZRLuDmwr3tj-8oOv6EfLAxV0dJAPtUT203Iw";
+    private static final String SERVICE_AUTHORISATION_VALUE = BEARER + TOKEN;
+
+
+    @Test
+    void shouldSuccessfullyUploadDocument() throws Exception {
+
+        stubDocumentUrl();
+        stubUploadDocument();
+        stubDocumentManagementUploadDocument();
+
+        MockMultipartFile firstFile =
+            new MockMultipartFile("files", "filename.txt",
+                                  "text/plain", "some xml".getBytes());
+        MockMultipartFile secondFile =
+            new MockMultipartFile("data", "other-file-name.data",
+                                  "text/plain", "some other type".getBytes());
+        MockMultipartFile jsonFile =
+            new MockMultipartFile("json", "",
+                                  MediaType.APPLICATION_JSON_VALUE, "{\"json\": \"someValue\"}".getBytes());
+
+        MockMvc mockMvcBuilder
+            = MockMvcBuilders.webAppContextSetup(webApplicationContext).build();
+        mockMvcBuilder.perform(MockMvcRequestBuilders.multipart(MAIN_URL)
+                            .file(firstFile)
+                            .file(secondFile)
+                            .file(jsonFile)
+                            .headers(createHttpHeaders(SERVICE_NAME_XUI_WEBAPP))
+                            .param(CLASSIFICATION_KEY, CLASSIFICATION)
+                            .param(CASE_TYPE_ID_KEY, CASE_TYPE_ID)
+                            .param(JURISDICTION_ID_KEY, JURISDICTION_ID)
+                            .contentType(MediaType.MULTIPART_FORM_DATA_VALUE))
+            .andExpect(status().isOk());
+    }
+
+
+    @Test
+    void shouldSuccessfullyGetDocumentByDocumentId() throws Exception {
+        String metaDataJsonExpression = "$.metadata.";
+
+        StoredDocumentHalResource storedDocumentResource = getStoredDocumentResource();
+
+        stubDocumentUrl();
+        stubGetDocumentMetaData(storedDocumentResource);
+
+        mockMvc.perform(get(MAIN_URL + "/" +  DOCUMENT_ID)
+                            .headers(createHttpHeaders(SERVICE_NAME_XUI_WEBAPP)))
+            .andExpect(status().isOk())
+            .andExpect(jsonPath(metaDataJsonExpression + CASE_ID_KEY, is(CASE_ID)))
+            .andExpect(jsonPath(metaDataJsonExpression + CASE_TYPE_ID_KEY, is(CASE_TYPE_ID)))
+            .andExpect(jsonPath(metaDataJsonExpression + JURISDICTION_ID_KEY, is(JURISDICTION_ID)))
+            .andExpect(jsonPath("$._links.self.href",
+                                is("http://localhost" + MAIN_URL + "/" + DOCUMENT_ID)));
```

Comment:
Can you assert logaAudit for all the tests please. See [here](https://github.com/hmcts/cpo-case-payment-orders-api/blob/master/src/integrationTest/java/uk/gov/hmcts/reform/cpo/controllers/CasePaymentOrdersControllerIT.java#L313)